src/common/cmdparse.h

```diff
@@ -96,9 +97,9 @@ bool cmd_getval(
 
 template <typename T>
 void
-cmd_putval(CephContext *cct, cmdmap_t& cmdmap, const std::string& k, const T& val)
+cmd_putval(CephContext *cct, cmdmap_t& cmdmap, std::string_view k, const T& val)
 {
-  cmdmap[k] = val;
+  cmdmap.insert_or_assign(std::string{k}, val);
```

Comment:
This might impose some extra work, I guess. Some `std::string` implementations use shared buffer techniques (COW) to deduplicate the memory while the `std::string` constructor, IIUC, copies `k.size()` bytes from `k.data()` into a fresh region.

However, I don't expect `cmd_putval()` nor even `cmd_getval()` to be on hot paths, and thus the performance shouldn't be really a concern.